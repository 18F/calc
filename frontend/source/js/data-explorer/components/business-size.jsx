import React from 'react';
import { connect } from 'react-redux';

import { makeOptions } from './util';
import { setBusinessSize as setBusinessSizeAction } from '../actions';
import { BUSINESS_SIZE_LABELS } from '../constants';

function BusinessSize({ idPrefix, size, setSize }) {
  const id = `${idPrefix}business_size`;
  const handleChange = e => { setSize(e.target.value); };

  return (
    <div className="filter filter-business_size">
      <label htmlFor={id}>Business size:</label>
      <select id={id} name="business_size"
              value={size} onChange={handleChange}>
        {makeOptions(BUSINESS_SIZE_LABELS)}
      </select>
    </div>
  );
}

BusinessSize.propTypes = {
  size: React.PropTypes.string.isRequired,
  setSize: React.PropTypes.func.isRequired,
  idPrefix: React.PropTypes.string,
};

BusinessSize.defaultProps = {
  idPrefix: '',
};

export default connect(
  state => ({ size: state.business_size }),
  { setSize: setBusinessSizeAction }
)(BusinessSize);
