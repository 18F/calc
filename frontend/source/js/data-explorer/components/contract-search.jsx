import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { setSchedule as setScheduleAction } from '../actions';
import { setQueryBy as setQueryByAction } from '../actions';
import { QUERY_BY_CONTRACT} from '../constants';

export function ContractNum({ setSchedule, setQueryBy }) {
  const handleClick = (e) => {
    // unset selectedSchedule if it has been previously set
    setSchedule('');
    setQueryBy(e.target.value);
  }

  return (
    <button
      type="button"
      id={QUERY_BY_CONTRACT}
      name={QUERY_BY_CONTRACT}
      value={QUERY_BY_CONTRACT}
      onClick={handleClick}
    >
      Search by contract number
    </button>
  );
}

ContractNum.propTypes = {
  setQueryBy: PropTypes.func.isRequired,
};

export default connect(
  state => ({
    selectedSchedule: state.schedule,
    queryBy: state.query_by,
  }),
  {
    setSchedule: setScheduleAction,
    setQueryBy: setQueryByAction,
  },
)(ContractNum);
