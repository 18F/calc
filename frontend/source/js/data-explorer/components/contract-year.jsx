import React from 'react';
import { connect } from 'react-redux';

import Tooltip from './tooltip';
import { setContractYear as setContractYearAction } from '../actions';
import {
  CONTRACT_YEAR_CURRENT,
  CONTRACT_YEAR_1,
  CONTRACT_YEAR_2,
  CONTRACT_YEAR_LABELS,
} from '../constants';

function ContractYear({ idPrefix, contractYear, setContractYear }) {
  const idCurrent = `${idPrefix}current-year`;
  const id1 = `${idPrefix}one-year-out`;
  const id2 = `${idPrefix}two-years-out`;
  const setYear = year => () => { setContractYear(year); };

  return (
    <div className="filter contract-year">
      <fieldset className="fieldset-inputs">

        <legend>Contract year:</legend>
        <a href="#" className="tooltip filter-more-info">
           <Tooltip text="All five years of pricing are available in the export">
             What's this?
           </Tooltip>
        </a>
        <h3 className="sr-only">Contract Year</h3>

        <ul className="contract-year-block">
          <li className="contract-list-item">
            <label htmlFor={idCurrent} className="radio">
              <input id={idCurrent} type="radio"
                     checked={contractYear === CONTRACT_YEAR_CURRENT}
                     onChange={setYear(CONTRACT_YEAR_CURRENT)}
                     name="contract-year" value={CONTRACT_YEAR_CURRENT}
                     tabIndex="0" />
              <span tabIndex="-1"
                    className="contract-current-year-span checkbox-focus"
                    aria-hidden="true">Current</span>
              <span className="sr-only">
                {CONTRACT_YEAR_LABELS[CONTRACT_YEAR_CURRENT]}
              </span>
            </label>
          </li>
          <li className="contract-list-item">
            <label htmlFor={id1} className="radio">
              <input id={id1} type="radio"
                     checked={contractYear === CONTRACT_YEAR_1}
                     onChange={setYear(CONTRACT_YEAR_1)}
                     name="contract-year" value={CONTRACT_YEAR_1}
                     tabIndex="0" />
              <span tabIndex="-1"
                    className="checkbox-focus"
                    aria-hidden="true">+1</span>
              <span className="sr-only">
                {CONTRACT_YEAR_LABELS[CONTRACT_YEAR_1]}
              </span>
            </label>
          </li>
          <li className="contract-list-item">
            <label htmlFor={id2} className="radio">
              <input id={id2} type="radio"
                     name="contract-year" value={CONTRACT_YEAR_2}
                     checked={contractYear === CONTRACT_YEAR_2}
                     onChange={setYear(CONTRACT_YEAR_2)}
                     tabIndex="0" />
              <span tabIndex="-1"
                    className="contract-last-year-span checkbox-focus"
                    aria-hidden="true">+2</span>
              <span className="sr-only">
                {CONTRACT_YEAR_LABELS[CONTRACT_YEAR_2]}
              </span>
            </label>
          </li>
        </ul>
      </fieldset>
    </div>
  );
}

ContractYear.propTypes = {
  contractYear: React.PropTypes.string.isRequired,
  setContractYear: React.PropTypes.func.isRequired,
  idPrefix: React.PropTypes.string,
};

ContractYear.defaultProps = {
  idPrefix: '',
};

export default connect(
  state => ({ contractYear: state['contract-year'] }),
  { setContractYear: setContractYearAction }
)(ContractYear);
