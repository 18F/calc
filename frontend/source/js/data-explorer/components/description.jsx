import React from 'react';
import { connect } from 'react-redux';

import {
  MIN_EXPERIENCE,
  MAX_EXPERIENCE,
  EDU_LABELS,
  SITE_LABELS,
  BUSINESS_SIZE_LABELS,
  SCHEDULE_LABELS,
} from '../constants';

import { formatCommas } from '../util';

function Filter({
  extraClassName,
  label,
  children,
}) {
  let className = 'filter';

  if (extraClassName) {
    className += ` ${extraClassName}`;
  }

  return (
    <span className={className}>
      {label ? `${label}: ` : null}
      <a className="focus-input" href="#">
        {children}
      </a>
    </span>
  );
}

Filter.propTypes = {
  extraClassName: React.PropTypes.string,
  label: React.PropTypes.string,
  children: React.PropTypes.any.isRequired,
};

function Description({
  shownResults,
  totalResults,
  minExperience,
  maxExperience,
  education,
  site,
  businessSize,
  schedule,
  laborCategory,
}) {
  let results = ' results ';
  const filtersClasses = ['filters'];
  const filters = [];

  if (laborCategory) {
    filters.push(<Filter key="lab">{laborCategory}</Filter>);
  }

  if (education.length) {
    filters.push(
      <Filter key="edu" label="education level"
              extraClassName="education-filter">
        {education.map(x => EDU_LABELS[x]).join(', ')}
      </Filter>
    );
  }

  if (minExperience !== MIN_EXPERIENCE ||
      maxExperience !== MAX_EXPERIENCE) {
    filters.push(
      <Filter key="exp" label="experience">
        {minExperience} - {maxExperience} years
      </Filter>
    );
  }

  if (site) {
    filters.push(
      <Filter key="sit" label="worksite">
        {SITE_LABELS[site]}
      </Filter>
    );
  }

  if (businessSize) {
    filters.push(
      <Filter key="bus" label="business size">
        {BUSINESS_SIZE_LABELS[businessSize]}
      </Filter>
    );
  }

  if (schedule) {
    filters.push(
      <Filter key="sch" label="schedule">
        {SCHEDULE_LABELS[schedule]}
      </Filter>
    );
  }

  if (filters.length) {
    results += 'with ';
  } else {
    filtersClasses.push('hidden');
  }

  // TODO: The original version of this faded-in (but never out)
  // whenever it changed. We might want to do that too, or choose
  // a different animation.

  // TODO: The original version of this didn't show the filters on
  // the very first request (it also kept track of how many requests
  // had been made) but I'm not sure why that was good for usability,
  // so I've left the functionality out for now. We can bring it back
  // if needed.

  return (
    <p className="">
      {`Showing ${formatCommas(shownResults)} of `}
      <span className="total">{formatCommas(totalResults)}</span>
      {results}
      <span className={filtersClasses.join(' ')}>
        {filters}
      </span>
    </p>
  );
}

Description.propTypes = {
  shownResults: React.PropTypes.number.isRequired,
  totalResults: React.PropTypes.number.isRequired,
  minExperience: React.PropTypes.number.isRequired,
  maxExperience: React.PropTypes.number.isRequired,
  education: React.PropTypes.array.isRequired,
  site: React.PropTypes.string.isRequired,
  businessSize: React.PropTypes.string.isRequired,
  schedule: React.PropTypes.string.isRequired,
  laborCategory: React.PropTypes.string.isRequired,
};

function mapStateToProps(state) {
  return {
    shownResults: state.rates.data.results.length,
    totalResults: state.rates.data.count,
    minExperience: state.min_experience,
    maxExperience: state.max_experience,
    education: state.education,
    site: state.site,
    businessSize: state.business_size,
    schedule: state.schedule,
    laborCategory: state.q,
  };
}

export default connect(mapStateToProps)(Description);
