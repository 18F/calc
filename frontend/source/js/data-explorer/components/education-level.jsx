/* global document */

import React from 'react';
import { connect } from 'react-redux';

import {
  autobind,
  handleEnterOrSpace,
} from '../util';

import {
  EDU_LABELS,
} from '../constants';

import {
  toggleEducationLevel,
} from '../actions';

// TODO: We could just use jQuery for this, but I wanted to decouple
// the new React code from jQuery as much as possible for now.
function elementContains(container, contained) {
  let target = contained.parentNode;

  while (target) {
    if (target === container) {
      return true;
    }
    target = target.parentNode;
  }
  return false;
}

/**
 * The following logic was created to mimic the following
 * legacy jQuery behavior:
 *
 *   Dropdown with Multiple checkbox select with jQuery - May 27, 2013
 *   (c) 2013 @ElmahdiMahmoud
 *   license: http://www.opensource.org/licenses/mit-license.php
 */

class EducationLevel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
    };
    autobind(this, ['handleToggleMenu', 'handleDocumentClick']);
  }

  handleDocumentClick(e) {
    if (!this.state.expanded) {
      return;
    }

    // Whenever users click outside of our dropdown, hide it.
    if (!elementContains(this.dropdownEl, e.target)) {
      this.setState({ expanded: false });
    }
  }

  handleToggleMenu(e) {
    // TODO: The original jQuery logic used .slideDown() here, it'd
    // be nice if we could use some sort of transition too.

    e.preventDefault();
    this.setState({
      expanded: !this.state.expanded,
    });
  }

  handleCheckboxClick(level) {
    this.props.toggleEducationLevel(level);
  }

  componentDidMount() {
    document.addEventListener('click', this.handleDocumentClick);
    document.addEventListener('focus', this.handleDocumentClick, true);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleDocumentClick);
    document.removeEventListener('focus', this.handleDocumentClick, true);
  }

  render() {
    const levels = this.props.levels;
    const idPrefix = this.props.idPrefix;
    const ddStyle = this.state.expanded ? { display: 'block' } : null;
    const inputs = Object.keys(EDU_LABELS).map(value => {
      const id = idPrefix + value;

      return (
        <li key={value}>
          <input
            id={id}
            type="checkbox"
            value={value}
            checked={levels.indexOf(value) >= 0}
            onChange={this.handleCheckboxClick.bind(this, value)}
            name="education" />
          <label htmlFor={id}>{EDU_LABELS[value]}</label>
        </li>
      );
    });
    let linkContent;

    if (levels.length === 0) {
      linkContent = (
        <span className="eduSelect">Select
          <span className="sr-only"> to reveal Education Level options</span>
        </span>
      );
    } else {
      const selectedLevels = levels.map(value => {
        const label = EDU_LABELS[value];
        return <span key={value} title={label}>{label}</span>;
      });
      linkContent = (
        <div className="multiSel">{selectedLevels}</div>
      );
    }

    return (
      <div>
        <label>Education level:</label>
        <dl className="dropdown"
            ref={(el) => { this.dropdownEl = el; } }>
          <dt>
            <a href="#" onClick={this.handleToggleMenu}
               role="button"
               aria-expanded={this.state.expanded.toString()}
               onKeyDown={handleEnterOrSpace(this.handleToggleMenu)}
               className={levels.length === 0 ? '' : 'filter_active'}>
              {linkContent}
            </a>
          </dt>

          <dd>
            <div className="multiSelect">
              <fieldset>
                <legend className="sr-only">Education level:</legend>
                <ul style={ddStyle}>
                  {inputs}
                </ul>
              </fieldset>
            </div>
          </dd>
        </dl>
      </div>
    );
  }
}

EducationLevel.propTypes = {
  levels: React.PropTypes.array.isRequired,
  idPrefix: React.PropTypes.string,
  toggleEducationLevel: React.PropTypes.func.isRequired,
};

EducationLevel.defaultProps = {
  idPrefix: 'education-level-',
};

export default connect(
  state => ({ levels: state.education }),
  { toggleEducationLevel }
)(EducationLevel);
