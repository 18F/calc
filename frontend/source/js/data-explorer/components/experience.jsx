/* global $ */

import React from 'react';
import { connect } from 'react-redux';

import {
  setExperience,
} from '../actions';

import {
  autobind,
} from '../util';

import {
  MIN_EXPERIENCE,
  MAX_EXPERIENCE,
} from '../constants';

class Experience extends React.Component {
  constructor(props) {
    super(props);
    autobind(this, ['onSliderSlide', 'onSliderChange']);
    this.state = {
      isSliding: false,
      sliderVal: null,
    };
  }

  componentDidMount() {
    // We're currently using an old version of noUiSlider (7.0.10). Its
    // documentation can only be seen via archive.org:
    //
    // https://web.archive.org/web/20141224210103/http://refreshless.com/nouislider/

    $(this.sliderEl).noUiSlider({
      start: [
        this.props.min,
        this.props.max,
      ],
      step: 1,
      connect: true,
      range: {
        min: MIN_EXPERIENCE,
        max: MAX_EXPERIENCE,
      },
    });
    $(this.sliderEl).on({
      slide: this.onSliderSlide,
      change: this.onSliderChange,
    });
  }

  componentDidUpdate() {
    if (!this.state.isSliding) {
      const [min, max] = this.getSliderVal();

      if (min !== this.props.min || max !== this.props.max) {
        $(this.sliderEl).val([this.props.min, this.props.max]);
      }
    }
  }

  getSliderVal() {
    return $(this.sliderEl).val().map(x => parseInt(x, 10));
  }

  onSliderSlide() {
    $('.noUi-horizontal .noUi-handle', this.rootEl).addClass('filter_focus');

    this.setState({
      isSliding: true,
      sliderVal: this.getSliderVal(),
    });
  }

  onSliderChange() {
    $('.noUi-horizontal .noUi-handle', this.rootEl)
      .removeClass('filter_focus');

    const [min, max] = this.getSliderVal();

    this.setState({
      isSliding: false,
      sliderVal: null,
    });

    if (min !== this.props.min) {
      this.props.dispatch(setExperience('min', min));
    }
    if (max !== this.props.max) {
      this.props.dispatch(setExperience('max', max));
    }
  }

  makeOptions(min, max) {
    const options = [];

    for (let i = min; i <= max; i++) {
      options.push(<option key={i} value={i}>{i}</option>);
    }

    return options;
  }

  render() {
    let min = this.props.min;
    let max = this.props.max;

    if (this.state.isSliding) {
      [min, max] = this.state.sliderVal;
    }

    const minId = `${this.props.idPrefix}min`;
    const maxId = `${this.props.idPrefix}max`;
    const classes = ['select-small'];
    const minClasses = classes.concat(min === MIN_EXPERIENCE ? [] : [
      'filter_active',
    ]);
    const maxClasses = classes.concat(max === MAX_EXPERIENCE ? [] : [
      'filter_active',
    ]);
    const onChange = type => e => {
      this.props.dispatch(setExperience(type, parseInt(e.target.value, 10)));
    };

    return (
      <div className="filter" ref={(el) => { this.rootEl = el; }}>
        <fieldset>
          <legend>
            Experience:
          </legend>
          <div className="slider" ref={(el) => { this.sliderEl = el; }} />
          <div className="experience_range">
            <label htmlFor={minId} className="sr-only">Minimum Years</label>
            <select id={minId} value={min} name="min_experience"
                    onChange={onChange('min')}
                    className={minClasses.join(' ')}>
              {this.makeOptions(MIN_EXPERIENCE, max)}
            </select>
            {' - '}
            <label htmlFor={maxId} className="sr-only">Maximum Years</label>
            <select id={maxId} value={max} name="max_experience"
                    onChange={onChange('max')}
                    className={maxClasses.join(' ')}>
              {this.makeOptions(min, MAX_EXPERIENCE)}
            </select>
            {' years'}
          </div>
        </fieldset>
      </div>
    );
  }
}

Experience.propTypes = {
  min: React.PropTypes.number.isRequired,
  max: React.PropTypes.number.isRequired,
  dispatch: React.PropTypes.func.isRequired,
  idPrefix: React.PropTypes.string,
};

Experience.defaultProps = {
  idPrefix: 'experience-',
};

function mapStateToProps(state) {
  return {
    min: state.min_experience,
    max: state.max_experience,
  };
}

export default connect(mapStateToProps)(Experience);
