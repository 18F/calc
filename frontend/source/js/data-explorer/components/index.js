export { default as Description } from './description';
export { default as Highlights } from './highlights';
export { default as Histogram } from './histogram';
export { default as ProposedPrice } from './proposed-price';
export { default as EducationLevel } from './education-level';
export { default as Experience } from './experience';
export { default as ExportData } from './export-data';
export { default as ResultsTable } from './results-table';
export { default as Schedule } from './schedule';
export { default as ContractYear } from './contract-year';
export { default as QueryType } from './query-type';
export { default as Site } from './site';
export { default as BusinessSize } from './business-size';
