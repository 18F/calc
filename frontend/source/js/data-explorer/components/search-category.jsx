import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { setQueryBy as setQueryByAction } from '../actions';
import Schedule from './schedule';
import Vendor from './vendor-search';
import ContractNum from './contract-search';
import {
  QUERY_BY_SCHEDULE,
  QUERY_BY_VENDOR,
  QUERY_BY_CONTRACT,
  DEFAULT_QUERY_BY
} from '../constants';
import { scheduleLabels } from '../schedule-metadata';

import {
  autobind,
  handleEnterOrSpace,
} from '../util';


export class SearchCategory extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
    };
    autobind(this, ['toggleDropdown', 'closeMenuOnClick', 'createButtonText', ]);
  }

  toggleDropdown() {
    this.setState({
      expanded: !this.state.expanded, /* eslint-disable-line react/no-access-state-in-setstate */
    });
  }

  closeMenuOnClick() {
    this.setState({
      expanded: false,
    });
  }

  createButtonText() {
    const {
      selectedSchedule,
      queryBy,
      queryBySchedule,
      queryByVendor,
      queryByContract
    } = this.props;
    let searchSummary;
    if (queryBy === queryBySchedule) {
      let extraContext;
      let allSchedsLabel = `${Object.keys(scheduleLabels).length} contract vehicles`;

      if (!this.state.expanded) {
        extraContext =(
          <span>
            in
            {' '}
            { scheduleLabels[selectedSchedule] || allSchedsLabel }
          </span>
        );
      };

      searchSummary = (
        <div>
          <strong>
            Search labor categories
          </strong>
          { extraContext }
        </div>
      );
    } else if (queryBy == queryByVendor) {
      searchSummary = (
        <strong>Search by vendor name</strong>
      );
    } else if (queryBy == queryByContract) {
      searchSummary = (
        <strong>Search by contract number</strong>
      );
    }
    return searchSummary;
  }

  render() {
    return (
      <div className="html-dropdown">
        <button
          type="button"
          className="html-dropdown__trigger"
          aria-controls="data-explorer__search-category" /* eslint-disable-line jsx-a11y/aria-proptypes */
          onClick={this.toggleDropdown}
          onKeyDown={handleEnterOrSpace(this.toggleDropdown)}
          aria-expanded={this.state.expanded}
        >
          { this.createButtonText() }
        </button>
        {/* setting a key event on this div makes it impossible to select
          * an option with the keyboard.
          * TODO: fix this after contract/vendor name are added and the HTML is in final form. */}
        <div /* eslint-disable-line max-len, jsx-a11y/click-events-have-key-events, jsx-a11y/interactive-supports-focus */
          className="html-dropdown__choices"
          id="data-explorer__search-category"
          onClick={this.closeMenuOnClick}
          aria-hidden={!this.state.expanded}
          role="menu"
        >
          {/* TODO: "all schedules" no longer works as a selector when contract or vendor are selected? */}
          <Schedule />
          <h3>Search vendors and contracts</h3>
          {/* TODO: Add checkmarks to these as background images on buttons when selected */}
          <Vendor />
          <ContractNum />
        </div>
      </div>
    );
  }
}

SearchCategory.propTypes = {
  selectedSchedule: PropTypes.string,
  queryBySchedule: PropTypes.string.isRequired,
  queryByVendor: PropTypes.string.isRequired,
  queryByContract: PropTypes.string.isRequired,
};

SearchCategory.defaultProps = {
  selectedSchedule: '',
  queryBySchedule: QUERY_BY_SCHEDULE,
  queryByVendor: QUERY_BY_VENDOR,
  queryByContract: QUERY_BY_CONTRACT,
};

export default connect(
  state => ({
    selectedSchedule: state.schedule,
    queryBy: state.query_by,
  })
)(SearchCategory);
