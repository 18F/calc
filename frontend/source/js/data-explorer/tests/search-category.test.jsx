import { SearchCategory } from '../components/search-category';
import makeSetup from './testSetup';

const defaultProps = {
  selectedSchedule: '',
};

const setup = makeSetup(SearchCategory, defaultProps, { wrapperOnly: true });

describe('<SearchCategory>', () => {
  it('renders correctly', () => {
    const { wrapper } = setup();
    const trigger = wrapper.find('.html-dropdown__trigger');
    expect(trigger.exists()).toBeTruthy();
    const dropdown = wrapper.find('.html-dropdown__choices');
    expect(dropdown.exists()).toBeTruthy();
  });
});
