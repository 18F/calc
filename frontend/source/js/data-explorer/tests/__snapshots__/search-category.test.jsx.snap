// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCategory> matches snapshot 1`] = `
<div
  className="html-dropdown"
>
  <button
    aria-controls="data-explorer__search-category"
    aria-expanded={false}
    className="html-dropdown__trigger"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  />
  <div
    aria-hidden={true}
    className="html-dropdown__choices"
    id="data-explorer__search-category"
    onClick={[Function]}
    role="menu"
  >
    <h3>
      Search for labor categories
    </h3>
    <Connect(Schedule) />
    <h3>
      Search for vendors and contracts
    </h3>
    <Connect(Vendor) />
    <Connect(ContractNum) />
  </div>
</div>
`;
